bundle/config/mutator/python/override.go

```diff
@@ -0,0 +1,313 @@
+package python
+
+import (
+	"fmt"
+
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/merge"
+)
+
+// overrideMode restricts which changes are allowed during the override process
+type overrideMode string
+
+const OverrideModeAppend overrideMode = "append"
+const OverrideModeAppendOrModify overrideMode = "append-or-update"
+
+// isModified keeps track of whether a value has been modified during the override process
+type isModified struct {
+	value bool
+}
```

Comment:
If this is only a bool then it should be typed as a bool.

Reply:
I didn't add bool intentionally because it's quite error-prone because it's easy to invert boolean values. So something like enum works better. We can also want to collect more information of what has been changed as a part of this struct.